include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -2498,6 +2498,54 @@ __pattern_sort_by_key(__parallel_tag<_IsVector>, _ExecutionPolicy&& __exec, _Ran
     });
 }
 
+//------------------------------------------------------------------------
+// stable_sort_by_key
+//------------------------------------------------------------------------
+
+template <class _Tag, typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _RandomAccessIterator2,
+          typename _Compare>
+void
+__pattern_stable_sort_by_key(_Tag, _ExecutionPolicy&& __exec, _RandomAccessIterator1 __keys_first,
+                             _RandomAccessIterator1 __keys_last, _RandomAccessIterator2 __values_first,
+                             _Compare __comp) noexcept
+{
+    static_assert(__is_serial_tag_v<_Tag> || __is_parallel_forward_tag_v<_Tag>);
+
+    auto __beg = oneapi::dpl::make_zip_iterator(__keys_first, __values_first);
+    auto __end = __beg + (__keys_last - __keys_first);
+    auto __cmp_f = [__comp](const auto& __a, const auto& __b) {
+        return __comp(::std::get<0>(__a), ::std::get<0>(__b));
+    };
+
+    ::std::stable_sort(__beg, __end, __cmp_f);
+}
+
+template <typename _IsVector, typename _ExecutionPolicy, typename _RandomAccessIterator1,
+          typename _RandomAccessIterator2, typename _Compare>
+void
+__pattern_stable_sort_by_key(__parallel_tag<_IsVector>, _ExecutionPolicy&& __exec, _RandomAccessIterator1 __keys_first,
+                             _RandomAccessIterator1 __keys_last, _RandomAccessIterator2 __values_first, _Compare __comp)
+{
+    static_assert(
```

Comment:
I would suggest omitting such type requirement checks with startic_assert, due to
1) a user has got an error directly in a code place where such requirement needs.
2) concept and requires are upcoming... 

I know that such checks were before this PR... But, it was for ensuring that we didn't miss a some special case after "move_constructible" had been removed from algo signature.

Reply:
Done